### PR TITLE
15 kafka

### PR DESCRIPTION
**Pull request by liel-almog**

### CODE REVIEW

File: backend/app/core/config.py [hunk 1]
```diff
@@ -3,11 +3,15 @@
...
+        env_ignore_empty=True,
+        extra="ignore",
     )
 
     CONNECTION_STRING: str
     DB_NAME: str
+    KAFKA_BROKERS: list[str]
 
 
 settings = Settings()
```
Comment (MikeCr4ft) [inline backend/app/core/config.py:17]: from dotenv import load_dotenv

load_dotenv()
to load the env file

File: vip-consumer/core/consumer.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+CONSUMER_GROUP = "vip_custom_requests_group"
+
+
+running = True
+
+
+async def msg_process(msg: ConsumerRecord) -> UpdateResult:
+    custom_request = CustomRequest(**json.loads(msg.value))
+    logger.info(f"Processing custom request: {custom_request}")
+    await asyncio.sleep(3)
```
Comment (MikeCr4ft) [inline vip-consumer/core/consumer.py:28]: why?

File: vip-consumer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.12-slim
+
+WORKDIR /code
+
+COPY ./requirements.txt /code/requirements.txt
+
+RUN pip install --no-cache-dir --upgrade -r /code/requirements.txt
+
+COPY . .
+COPY .env .env
```
Comment (MikeCr4ft) [inline vip-consumer/Dockerfile:10]: you can't copy the env file,  its best practise to inject the env file when building

File: vip-consumer/core/confuelnt_consumer.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import sys
+from confluent_kafka import Consumer, KafkaError, KafkaException
+
+MIN_COMMIT_COUNT = 10
+TOPICS = ["vip_custom_request"]
```
Comment (MikeCr4ft) [inline vip-consumer/core/confuelnt_consumer.py:5]: why isnt this in the env file?

File: vip-consumer/core/db.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+async def close_mongo():
+    global _client
+
+    try:
+        _client.close()
+        logger.info("Closed connection to mongoDB")
+    except Exception as _e:
+        logger.error("Unable to closed connection to mongoDB")
```
Comment (MikeCr4ft) [inline vip-consumer/core/db.py:40]: why not do everything in a class instead of using this shit global

File: vip-consumer/core/consumer.py (hunk 2 above)
Comment (liel-almog) [reply vip-consumer/core/consumer.py:28]: This is what gali said

File: vip-consumer/core/confuelnt_consumer.py (hunk 4 above)
Comment (liel-almog) [reply vip-consumer/core/confuelnt_consumer.py:5]: This is something that does not change

File: backend/app/core/config.py (hunk 1 above)
Comment (liel-almog) [reply backend/app/core/config.py:17]: You can read the [docs](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#usage) this is how they do it.

File: vip-consumer/Dockerfile (hunk 3 above)
Comment (liel-almog) [reply vip-consumer/Dockerfile:10]: You inject the env file to the code and then use it inside the docker.

File: vip-consumer/core/db.py (hunk 5 above)
Comment (liel-almog) [reply vip-consumer/core/db.py]: Fixed in [commit](https://github.com/liel-almog/gala/pull/16/commits/6e3e645c83b84f971ffa1c523f377f0e5e59d5c4)

**Pull request merged**